### ISSUE TITLE
(Question) Return a compact version of a resource

### ISSUE THREAD

**Issue by kclay** [LartTyler/MHWDB-Docs#21]
So I see that you have added the set bonus, perfect timing I must say. But I wanted to know if it is possible to return a subset of the data.

Take https://mhw-db.com/armor/sets/111 for example.

It would be nice to have either of the following
- https://mhw-db.com/armor/sets?fields=id,name,rank,pieces.id,bonus
- https://mhw-db.com/armor/sets/111?fields=id,name,rank,pieces.id,bonus
- https://mhw-db.com/armor/sets/111?compact=true

The expected response for `111` set would be the following

```json
{
"id":111,
   "name":"Diablos Alpha",
   "rank":"high",
   "pieces":[487,488,489,490,491],
   "bonus":{  
      "id":12,
      "name":"Diablos Mastery",
      "ranks":[  
         {  
            "pieces":2,
            "skill":{  
               "id":315,
               "slug":"bludgeoner-rank-1",
               "level":1,
               "description":"Raises attack as your weapon loses sharpness. Also boosts ranged weapon melee attacks and odds of stunning.",
               "modifiers":[  

               ],
               "skill":116,
               "skillName":"Bludgeoner"
            }
         }
      ]
   }
}
```

Basically just removing the full Armor payload and just return the ids.

**Comment by LartTyler** [comment]
I've been wanting to add the ability to project certain fields, I just haven't had a chance to yet. The API is set up in a way that this would be easy enough to add, it's just a matter of prioritizing it. I'll keep you posted!

**Comment by LartTyler** [comment]
@kclay This has (finally) been added with [v1.12.0](https://github.com/LartTyler/MHWDB-API/releases/tag/1.12.0). Field projection can be done via the `p` query parameter, and full usage documentation can be found [here](https://docs.mhw-db.com/#projecting-results).
